packaging/build_packages_deb.sh

```diff
@@ -26,11 +26,21 @@ trap exit_error ERR
 URL="http://metadata/computeMetadata/v1/instance/attributes"
 GCS_PATH=$(curl -f -H Metadata-Flavor:Google ${URL}/daisy-outs-path)
 BASE_REPO=$(curl -f -H Metadata-Flavor:Google ${URL}/base-repo)
+REPO=$(curl -f -H Metadata-Flavor:Google ${URL}/repo)
+PULL_REF=$(curl -f -H Metadata-Flavor:Google ${URL}/pull-ref)
+
+
+echo "started build..."
+
+apt-get -y update && apt-get -y upgrade
+apt-get install -y git-core
+git clone --branch ${PULL_REF} "https://github.com/${BASE_REPO}/${REPO}.git"
```

Comment:
ex if you wanted to check out the code associated with a pull request, which is not yet merged to any real branch, you'd run something like:

```
git init
git fetch https://github.com/${BASE_REPO}/${REPO}.git
git fetch https://github.com/${BASE_REPO}/${REPO}.git pull/37/head:pull/37/head
git checkout pull/37/head
```

Reply:
at first; it is actually PULL_BASE_REF(not sure why i used PULL_REF; may have been just a typo); It  is the "Ref name of the base branch" according to prow. The example on the prow page, led me to the conclusion that it will be the branch name of the repo that we check'd out. i really did not test this part on the prow which is why i missed the ${PULL_REF} failure.

all we want here is to checkout to the branch that was used to send PR.

i think this might not work with BASE_REPO being set to GoogleCloudPLatform since the PR is sent from a forked repo.
https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md

Reply:
all of the refs will be relative to our repo, don't worry. when you file a PR, a special ref is created on your repo to point to the commit-to-be-merged. prow knows how to get these, and we need to support something somewhat similar for our builds. maybe we can use the clone util from prow directly, or with slight modification?

anyway here's an example of what's set in a PR for refs:
```
  refs:
    base_link: https://github.com/GoogleCloudPlatform/guest-logging-go/commit/5d4770fc537a2976761bcc1a78a3c41085246284
    base_ref: master
    base_sha: 5d4770fc537a2976761bcc1a78a3c41085246284
    org: GoogleCloudPlatform
    pulls:
    - author: hopkiw
      author_link: https://github.com/hopkiw
      commit_link: https://github.com/GoogleCloudPlatform/guest-logging-go/pull/3/commits/f2714eb104c424b90ecb0b9e07a87e49f4ab1b68
      link: https://github.com/GoogleCloudPlatform/guest-logging-go/pull/3
      number: 3
      sha: f2714eb104c424b90ecb0b9e07a87e49f4ab1b68
```

Reply:
played around with git PR refs.
I made the packaging scripts agnostic to prow job type. Based on the prow job type, the "pull ref" will be added to the daisy args.